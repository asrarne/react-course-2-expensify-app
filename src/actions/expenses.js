// import uuid from 'uuid';
// import moment from 'moment';
import dbExpenseRef from '../firebase/firebase';
import { set, push } from 'firebase/database';

//action generators
//ADD_EXPENSE
export const addExpense = (expense) => ({
    type: 'ADD_EXPENSE',
    expense
});

export const startAddExpense = (expenseData = {}) => {
    return (dispatch) => {
        const {
                description = '', 
                note = '', 
                amount = 0, 
                createdAt = 0
                } = expenseData;
        const expense = { description, note, amount, createdAt };

        const newExpenseRef = push(dbExpenseRef);

        return set(newExpenseRef, expense)
        .then(()=>{
            dispatch(addExpense({
                id: newExpenseRef.key,
                ...expense
            }));
        }).catch((e) => {
            console.log('startAddExpense failed with error: ', e);
        });
    };
}; 

//REMOVE_EXPENSE
export const removeExpense = ({id} = {}) => ({
    type: 'REMOVE_EXPENSE',
    id
});

//EDIT_EXPENSE
export const editExpense = (id, updates ) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
});