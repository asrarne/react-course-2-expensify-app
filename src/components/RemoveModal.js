import React from 'react';
import Modal from 'react-modal';


const RemoveModal = (props) => (
    <Modal
        isOpen={props.removeClicked}
        onRequestClose={props.handleRemoveNo}
        ariaHideApp={false}
        contentLabel="Do you want to remove expense?"
        closeTimeoutMS={200}
        className="modal"
    >
        <h3 className="modal__title"> Do you want to remove expense?</h3>
        <div className="modal__action">
        <button className="button" onClick={props.handleRemoveYes}>Yes</button>
        <button className="button" onClick={props.handleRemoveNo}>No</button>
        </div>
    </Modal>
);

export default RemoveModal;