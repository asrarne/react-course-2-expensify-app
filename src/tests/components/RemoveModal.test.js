import React from 'react';
import { shallow } from 'enzyme';
import RemoveModal from '../../components/RemoveModal';
import expenses from '../fixtures/expenses';

let handleRemoveYes, handleRemoveNo, removeClicked, wrapper;

beforeEach(() => {
    handleRemoveYes = jest.fn();
    handleRemoveNo = jest.fn();
    removeClicked = true;
    wrapper = shallow(<RemoveModal  
                                    removeClicked={removeClicked}
                                    handleRemoveYes={handleRemoveYes} 
                                    handleRemoveNo={handleRemoveNo} 
                                />);
});

test('should render RemoveModal correctly', () => {
    expect(wrapper).toMatchSnapshot();                                         
});

test('should handle remove expense if clicked Yes', () => {
    wrapper.find('button').at(0).simulate('click');
    expect(handleRemoveYes).toHaveBeenCalled();
});

test('should handle remove expense if clicked No', () => {
    wrapper.find('button').at(1).simulate('click');
    expect(handleRemoveNo).toHaveBeenCalled();
});