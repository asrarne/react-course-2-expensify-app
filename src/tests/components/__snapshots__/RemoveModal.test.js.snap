// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RemoveModal correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentElement={[Function]}
  contentLabel="Do you want to remove expense?"
  isOpen={true}
  onRequestClose={[MockFunction]}
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="modal__title"
  >
     Do you want to remove expense?
  </h3>
  <div
    className="content-container"
  >
    <div
      className="modal__action"
    >
      <button
        className="button"
        onClick={[MockFunction]}
      >
        Yes
      </button>
      <button
        className="button"
        onClick={[MockFunction]}
      >
        No
      </button>
    </div>
  </div>
</Modal>
`;
